src/plugins/auto/src/infer_request.cpp

```diff
@@ -97,8 +97,12 @@ void ov::auto_plugin::InferRequest::set_tensors_to_another_request(const SoAsync
         auto type = tensor->get_element_type();
         bool is_remote  = std::dynamic_pointer_cast<ov::IRemoteTensor>(tensor._ptr) ||
             std::dynamic_pointer_cast<ov::IRemoteTensor>(req->get_tensor(it)._ptr);
-        if (is_remote || req->get_tensor(it)->data(type) != tensor->data(type))
+        if (is_remote || req->get_tensor(it)->data(type) != tensor->data(type)) {
+            // Update tensor shape if it differs from the selected infer request tensor shape
+            if (req->get_tensor(it)->get_shape() != tensor->get_shape())
```

Comment:
let's add more strict check, only check the tensor is dynamic, and the tensor in hardware is not
and let's enable a test for it

Reply:
Updated. will enable test case soon.